Sources/GRPC/AsyncAwaitSupport/GRPCAsyncBidirectionalStreamingCall.swift

```diff
@@ -0,0 +1,163 @@
+      try await self.responseParts.initialMetadata.get()
+    }
+  }
+
+  /// The trailing metadata returned from the server.
+  ///
+  /// - Important: Awaiting this property will suspend until the responses have been consumed.
+  public var trailingMetadata: HPACKHeaders {
+    // swiftformat:disable:next redundantGet
+    get async throws {
+      try await self.responseParts.trailingMetadata.get()
+    }
+  }
+
+  /// The final status of the the RPC.
+  ///
+  /// - Important: Awaiting this property will suspend until the responses have been consumed.
+  public var status: GRPCStatus {
+    // swiftformat:disable:next redundantGet
+    get async {
+      // force-try acceptable because any error is encapsulated in a successful GRPCStatus future.
+      try! await self.responseParts.status.get()
+    }
+  }
+
+  private init(call: Call<Request, Response>) {
+    self.call = call
+    // Initialise `responseParts` with an empty response handler because we
+    // provide the responses as an AsyncSequence in `responseStream`.
+    self.responseParts = StreamingResponseParts(on: call.eventLoop) { _ in }
+
+    // Call and StreamingResponseParts are reference types so we grab a
+    // referecence to them here to avoid capturing mutable self in the  closure
+    // passed to the AsyncThrowingStream initializer.
+    //
+    // The alternative would be to declare the responseStream as:
+    // ```
+    // public private(set) var responseStream: AsyncThrowingStream<ResponsePayload>!
+    // ```
+    //
+    // UPDATE: Additionally we expect to replace this soon with an AsyncSequence
+    // implementation that supports yielding values from outside the closure.
+    let call = self.call
+    let responseParts = self.responseParts
+    let responseStream = AsyncThrowingStream(Response.self) { continuation in
+      call.invokeStreamingRequests { error in
+        responseParts.handleError(error)
+        continuation.finish(throwing: error)
+      } onResponsePart: { responsePart in
+        responseParts.handle(responsePart)
+        switch responsePart {
+        case let .message(response):
+          continuation.yield(response)
+        case .metadata:
+          break
+        case .end:
+          continuation.finish()
+        }
+      }
+    }
+    self.responses = .init(responseStream)
+  }
+
+  /// We expose this as the only non-private initializer so that the caller
+  /// knows that invocation is part of initialisation.
+  internal static func makeAndInvoke(call: Call<Request, Response>) -> Self {
+    Self(call: call)
+  }
+
+  // MARK: - Requests
+
+  /// Sends a message to the service.
+  ///
+  /// - Important: Callers must terminate the stream of messages by calling `sendEnd()`.
+  ///
+  /// - Parameters:
+  ///   - message: The message to send.
+  ///   - compression: Whether compression should be used for this message. Ignored if compression
+  ///     was not enabled for the RPC.
+  public func sendMessage(
```

Comment:
Where did we land with `GRPCAsync{Client,Bidirectional}StreamingCall` having a request-stream or similar?

Reply:
There's an unresolved thread above which I found a reply that I wrote but forgot to send. It's here: https://github.com/grpc/grpc-swift/pull/1243#discussion_r699191008.

I thought we were deferring this until we had the writer that supports backpressure?